### ISSUE TITLE
Sample Java TUS Server returns Location without url hostname, which breaks this flutter client

### ISSUE THREAD

**Issue by eldarj** [jjmutumi/tus_client#10]
As the title says, I've implemented the Java TUS Server sample, which upon uploading a file, returns a Location without the full-url, but just the endpoint, e.g. **/api/uploads/random-file-upload-id**

I just started looking into TUS, and not sure whether this is an issue on the server side or this flutter client implementtion, but the client breaks with an exception saying that the Location arg. i.e. resource URL doesn't contain the hostname/base.

I've inspected the client, and fixed this with prepending the Location url with the hostname, but as said, I'm not sure whether this is an issue on the server or client side impl.

Is there a way to set up the server to provide the full url for the resource? If not, this should be fixed on the flutter side.

My flutter dependencies:
```
  http: ^0.12.2
  tus_client: ^0.0.4
```

TUS Server dep:
```
        <dependency>
            <groupId>me.desair.tus</groupId>
            <artifactId>tus-java-server</artifactId>
            <version>1.0.0-2.0</version>
        </dependency>
```

How I temporarily fixed this is with passing the entire response within the **onComplete** callback - in which I fetch a custom header (e.g. X-Location) set manually on the server side.

Also posted on https://github.com/jjmutumi/tus_client/issues/10

**Comment by jjmutumi** [comment]
Looks similar to fix in PR #7. Please update to version **0.1.1** in your Flutter application and see if the issue is fixed:

```yaml
dependencies:
  tus_client: ^0.1.1
```

**Comment by jjmutumi** [comment]
https://github.com/jjmutumi/tus_client/releases/tag/v0.1.2 fixes this. Please update to version 0.1.2 in your application:

```
dependencies:
  tus_client: ^0.1.2
```
